### PR TITLE
adding visibility into current mirror state

### PR DESCRIPTION
**Pull request by heavycrystal**
breaks some protos and routes for clearer naming conventions, UI needs to be updated accordingly

removes `SetupComplete` and `SnapshotComplete` booleans in CDCFlowState in favour of a CurrentFlowState that expresses all states of a CDC workflow in a single variable.

### CODE REVIEW

**Comment by serprex** [comment]
If protos are breaking, could also remove deprecated fields

File: flow/e2e/snowflake/peer_flow_sf_test.go [hunk 1]
```diff
@@ -414,7 +414,7 @@ func (s *PeerFlowE2ETestSuiteSF) Test_Toast_SF(t *testing.T) {
 	// in a separate goroutine, wait for PeerFlowStatusQuery to finish setup
 	// and execute a transaction touching toast columns
 	go func() {
-		e2e.SetupCDCFlowStatusQuery(env, connectionGen)
+		e2e.SetupCDCFlowStateQuery(env, connectionGen)
```
Comment (serprex) [inline flow/e2e/snowflake/peer_flow_sf_test.go:417]: Moving renaming into a separate PR would make this easier to review

**Comment by serprex** [comment]
PR description should explain why cdc flow SetupComplete/SnapshotComplete bools are changed into `CurrentFlowState *protos.FlowStatus`

Was there a correctness issue? Was it not allowing enough info for pausing/terminating to be shown while in progress? Is there future improvements which this change is considering?

File: flow/cmd/handler.go [hunk 2]
```diff
@@ -350,30 +350,80 @@ func (h *FlowRequestHandler) ShutdownFlow(
...
 			shared.CDCFlowSignalName,
 			shared.NoopSignal,
 		)
+	} else if req.RequestedFlowState == protos.FlowStatus_STATUS_TERMINATED &&
+		(currState == protos.FlowStatus_STATUS_RUNNING.Enum() || currState == protos.FlowStatus_STATUS_PAUSED.Enum()) {
+		err = h.updateWorkflowStatus(ctx, workflowID, protos.FlowStatus_STATUS_TERMINATING.Enum())
+		if err != nil {
+			return nil, err
+		}
+		_, err = h.ShutdownFlow(ctx, &protos.ShutdownRequest{
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/cmd/handler.go:415]: this is to preserve existing behavior that waits till the workflow terminates before returning, while also ensuring we go from STATUS_TERMINATING -> STATUS_TERMINATED
